dependency/health_service.go

```diff
@@ -236,8 +239,13 @@ func ParseHealthServices(s ...string) (*HealthServices, error) {
 // "warning", and finally "passing". If there are no checks, the service will be
 // marked as "passing".
 func statusFromChecks(checks []*api.HealthCheck) (string, error) {
-	var passing, warning, unknown, critical bool
+	var passing, warning, unknown, critical, maintenance bool
 	for _, check := range checks {
+		if check.CheckID == NodeMaint || check.CheckID == ServiceMaint {
```

Comment:
service is formatted as <prefix>:<service id> so the service one needs to be a `StartsWith()`